packages/ui-components/src/components/toolbar.tsx

```diff
@@ -607,10 +625,57 @@ export class ToolbarButton extends ReactWidget {
   constructor(private props: ToolbarButtonComponent.IProps = {}) {
     super();
     addToolbarButtonClass(this);
+    this._enabled = props.enabled ?? true;
+    this._pressed = this._enabled === true ? props.pressed ?? false : false;
```

Comment:
Small simplification

```suggestion
    this._pressed = this._enabled && props.pressed;
```